File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/Java/recyclerview/GridKucingAdapter.java
package id.poncoe.latihanmasterandroidponcoe.Java.recyclerview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import java.util.ArrayList;
import id.poncoe.latihanmasterandroidponcoe.R;

public class GridKucingAdapter extends RecyclerView.Adapter<GridKucingAdapter.GridViewHolder> {
    private ArrayList<Kucing> listKucing;

    public GridKucingAdapter(ArrayList<Kucing> list) {
        this.listKucing = list;
    }

    private OnItemClickCallbackGrid onItemClickCallback;
    public void setOnItemClickCallback(OnItemClickCallbackGrid onItemClickCallback) {
        this.onItemClickCallback = onItemClickCallback;
    }

    @NonNull
    @Override
    public GridViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_grid_recyclerview, viewGroup, false);
        return new GridViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final GridViewHolder holder, int position) {

        /** Glide digunakan untuk memuat sebuah gambar, baik yang sudah Anda siapkan di drawable maupun berada di server */
        Glide.with(holder.itemView.getContext())
                .load(listKucing.get(position).getPhoto())
                .apply(new RequestOptions().override(350, 550).centerCropTransform())
                .into(holder.imgPhoto);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClickCallback.onItemClicked(listKucing.get(holder.getAdapterPosition()));
            }
        });

    }

    public interface OnItemClickCallbackGrid {
        void onItemClicked(Kucing data);
    }

    @Override
    public int getItemCount() {
        return listKucing.size();
    }

    class GridViewHolder extends RecyclerView.ViewHolder {
        ImageView imgPhoto;

        GridViewHolder(View itemView) {
            super(itemView);
            imgPhoto = itemView.findViewById(R.id.img_item_photo);
        }
    }
}

File: app/src/main/java/id/poncoe/latihanmasterandroidponcoe/Kotlin/recyclerview/GridKucingAdapter.kt
package id.poncoe.latihanmasterandroidponcoe.Kotlin.recyclerview

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import id.poncoe.latihanmasterandroidponcoe.Kotlin.recyclerview.Kucing
import id.poncoe.latihanmasterandroidponcoe.R

class GridKucingAdapter(val listKucing: ArrayList<Kucing>) : RecyclerView.Adapter<GridKucingAdapter.GridViewHolder>() {

    private lateinit var onItemClickCallback: OnItemClickCallbackGridKt

    fun setOnItemClickCallback(onItemClickCallback: OnItemClickCallbackGridKt) {
        this.onItemClickCallback = onItemClickCallback
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): GridViewHolder {
        val view: View = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_grid_recyclerview, viewGroup, false)
        return GridViewHolder(view)
    }

    override fun onBindViewHolder(holder: GridViewHolder, position: Int) {

        /** Glide digunakan untuk memuat sebuah gambar, baik yang sudah Anda siapkan di drawable maupun berada di server */
        Glide.with(holder.itemView.context)
                .load(listKucing[position].photo)
                .apply(RequestOptions().override(350, 550).centerCrop())
                .into(holder.imgPhoto)

        holder.itemView.setOnClickListener { onItemClickCallback.onItemClicked(listKucing[holder.adapterPosition]) }

    }

    interface OnItemClickCallbackGridKt {
        fun onItemClicked(data: Kucing)
    }

    override fun getItemCount(): Int {
        return listKucing.size
    }

    inner class GridViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imgPhoto: ImageView = itemView.findViewById(R.id.img_item_photo)
    }
}